packages/marko-web-omeda-identity-x/rapid-identify.js

```diff
@@ -0,0 +1,91 @@
+const gql = require('graphql-tag');
+const { get } = require('@parameter1/base-cms-object-path');
+
+
+const RAPID_IDENT = gql`
+  mutation RapidIdentityX($input: RapidCustomerIdentificationMutationInput!) {
+    result: rapidCustomerIdentification(input: $input) {
+      id
+      encryptedCustomerId
+    }
+  }
+`;
+
+const getAlpha3CodeFor = async (alpha2, identityX) => {
+  const { data } = await identityX.client.query({
+    query: gql`
+      query GetAlpha3Code($alpha2: String!) {
+        localeCountry(input: { code: $alpha2 }) {
+          id
+          alpha3
+        }
+      }
+    `,
+    variables: { alpha2 },
+  });
+  return get(data, 'localeCountry.alpha3');
+};
+
+/**
+ *
+ * @param {object} params
+ * @param {string} params.brandKey The Omeda brand key
+ * @param {number} params.productId The Omeda product ID to associate with the identification
+ * @param {object} params.appUser The IdentityX user
+ * @param {IdentityX} params.identityX The Marko web IdentityX service
+ * @param {ApolloClient} params.omedaGraphQL The Omeda GraphQL client
+ */
+module.exports = async ({
+  brandKey,
+  productId,
+  appUser,
+  identityX,
+  omedaGraphQL,
+} = {}) => {
+  const {
+    givenName,
+    familyName,
+    organization,
+    organizationTitle,
+    countryCode,
+    regionCode,
+    postalCode,
+  } = appUser;
+
+  let alpha3;
+  if (countryCode) {
+    // omeda requires and ISO alpha3 country code.
+    alpha3 = await getAlpha3CodeFor(countryCode, identityX);
+  }
+
+  const input = {
+    productId,
+    email: appUser.email,
+    ...(givenName && { firstName: givenName }),
+    ...(familyName && { lastName: familyName }),
+    ...(organization && { companyName: organization }),
+    ...(organizationTitle && { title: organizationTitle }),
+    ...(alpha3 && { countryCode: alpha3 }),
+    ...(regionCode && { regionCode }),
+    ...(postalCode && { postalCode }),
+  };
+  const { data } = await omedaGraphQL.mutate({
```

Comment:
Rather than doing 3 queries to two graph servers, why not do single query to the omeda graphql and have it kick off the external user id additions?

Reply:
Because the Omeda GraphQL server has nothing to do with converting IdentityX users. The Omeda API is "pure" in that it only handles direct Omeda operations. Something else needs to be responsible for converting the IdentityX user. That's what this is.